### DISCUSSION TITLE
So is the CUDA job scheduling related code in the kernel driver module?

### DISCUSSION THREAD

**Discussion by s66104444** [NVIDIA/open-gpu-kernel-modules#166] [General]
Hope the question is answered.

**Comment by aaronp24** [comment]
Do you have a specific question about scheduling? Scheduling is a complex topic so it would help to narrow down the question a bit.

In general though, the kernel driver is responsible for setting up access to the GPU but is not involved in kicking off work for the GPU to process. That's handled entirely in the user-mode drivers.

**Comment by s66104444** [comment]
So, the scheduling of the CUDA-launched kernel is handled by the user-mode driver?

**Reply by aaronp24** [reply]
Work is submitted to the GPU by the user-mode driver. Scheduling between jobs is handled by the hardware itself.

**Reply by s66104444** [reply]
Oh, I get it, thanks!

**Comment by johnnynunez** [comment]
Overclocking and undervolt on Linux could be made easier
